solidity/test/LiquidityPoolV1Converter.js

```diff
@@ -668,4 +675,171 @@ contract('LiquidityPoolV1Converter', accounts => {
             }
         }
     });
+
+    describe('recent average rate', () => {
+        const AVERAGE_RATE_PERIOD = duration.minutes(10);
+
+        let converter;
+        beforeEach(async () => {
+            converter = await initConverter(true, true, 5000, true);
+        });
+
+        const getExpectedAverageRate = (prevAverageRate, currentRate, timeElapsed) => {
+            if (timeElapsed.eq(new BN(0))) {
+                return prevAverageRate;
+            }
+
+            if (timeElapsed.gte(AVERAGE_RATE_PERIOD)) {
+                return currentRate;
+            }
+
+            const newAverageRateN = prevAverageRate.n.mul(currentRate.d).mul(AVERAGE_RATE_PERIOD.sub(timeElapsed))
+                .add(prevAverageRate.d.mul(currentRate.n).mul(timeElapsed));
+            const newAverageRateD = AVERAGE_RATE_PERIOD.mul(prevAverageRate.d).mul(currentRate.d);
+
+            return { n: newAverageRateN, d: newAverageRateD };
+        };
+
+        const expectRatesAlmostEqual = (rate, newRate) => {
+            const rate1 = Decimal(rate.n.toString()).div(Decimal(rate.d.toString()));
+            const rate2 = Decimal(newRate.n.toString()).div(Decimal(newRate.d.toString()));
+
+            const ratio = rate1.div(rate2);
+            expect(ratio.gte(0.999998)).to.be.true(`${ratio.toString()} is below MIN_RATIO`);
```

Comment:
Use strings, not (floating-point) numbers.

Reply:
done

Reply:
I think that you don't need `toString()` inside a `${}`.

Reply:
done